### ISSUE TITLE
Decouple emulated (HLE) backbuffer from actual

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#1118]
Luke had an idea about backbuffers/swap functions. We could create a host surface to serve as the default render target/backbuffer, and then in D3DDevice_Swap, blit that surface to the host backbuffer, fully seperating the xbox and host. This has the benefit of being able to treat the Xbox backbuffer like any other surface, and would also allow us to properly simulate the Xbox swapchain (multiple backbuffers, etc)

basically When the Xbox title calls SetRenderTarget with a value of NULL, we intercept it and make it render to our replacement surface.

It just gives us more control over the whole process

**Comment by PatrickvL** [comment]
In our UpdateOverlay patch we already have code that copies Xbox buffers to host. There's comparable code in our GetBackBuffer2 patch.
This code should be generalized, expanded so it can handle all formats and dimensions, and then used anywhere we need to update host textures with Xbox data.

**Comment by LukeUsher** [comment]
Done

**Comment by PatrickvL** [comment]
Done, true, except the part about generalising surface copies
